modules/data.remote/R/remote_process.R

```diff
@@ -0,0 +1,1054 @@
+  # collection validation to be implemented
+  if (!is.null(projection)) {
+    PEcAn.logger::severeifnot("projection should be of character type",
+                              is.character(projection))
+  }
+  if (!is.null(algorithm)) {
+    PEcAn.logger::severeifnot("algorithm should be of character type",
+                              is.character(algorithm))
+  }
+  if (!is.null(credfile)) {
+    PEcAn.logger::severeifnot("credfile should be of character type",
+                              is.character(credfile))
+  }
+  PEcAn.logger::severeifnot(
+    "Check if out_get_data is of character type and is not NULL",
+    is.character(out_get_data)
+  )
+  if (!is.null(out_process_data)) {
+    PEcAn.logger::severeifnot("out_process_data should be of character type",
+                              is.character(out_process_data))
+  }
+  
+  
+  dbcon <- PEcAn.DB::db.open(settings$database$bety)
+  on.exit(PEcAn.DB::db.close(dbcon), add = TRUE)
+  
+  # collection dataframe used to map Google Earth Engine collection names to their PEcAn specific names
+  collection_lut <- data.frame(
+    stringsAsFactors = FALSE,
+    original_name = c(
+      "LANDSAT/LC08/C01/T1_SR",
+      "COPERNICUS/S2_SR",
+      "NASA_USDA/HSL/SMAP_soil_moisture"
+    ),
+    pecan_code = c("l8", "s2", "smap")
+  )
+  getpecancode <- collection_lut$pecan_code
+  names(getpecancode) <- collection_lut$original_name
+  
+  if (source == "gee") {
+    collection = unname(getpecancode[collection])
+  }
+  
+  # construct raw file name
+  raw_file_name <-
+    construct_raw_filename(collection, siteid_short, scale, projection, qc)
+  
+  
+  # check if any data is already present in the inputs table
+  dbstatus <-
+    remotedata_db_check(
+      raw_file_name     = raw_file_name,
+      start             = start,
+      end               = end,
+      siteid            = siteid,
+      siteid_short      = siteid_short,
+      out_get_data      = out_get_data,
+      algorithm         = algorithm,
+      out_process_data  = out_process_data,
+      overwrite         = overwrite,
+      dbcon             = dbcon
+    )
+  
+  remotefile_check_flag  <- dbstatus[[1]]
+  start                  <- dbstatus[[2]]
+  end                    <- dbstatus[[3]]
+  stage_get_data         <- dbstatus[[4]]
+  write_raw_start        <- dbstatus[[5]]
+  write_raw_end          <- dbstatus[[6]]
+  raw_merge              <- dbstatus[[7]]
+  existing_raw_file_path <- dbstatus[[8]]
+  stage_process_data     <- dbstatus[[9]]
+  write_pro_start        <- dbstatus[[10]]
+  write_pro_end          <- dbstatus[[11]]
+  pro_merge              <- dbstatus[[12]]
+  input_file             <- dbstatus[[13]]
+  existing_pro_file_path <- dbstatus[[14]]
+  raw_check              <- dbstatus[[15]]
+  pro_check              <- dbstatus[[16]]
+  
+  # construct outdir path
+  outdir <-
+    file.path(outdir, paste(source, "site", siteid_short, sep = "_"))
```

Comment:
could be nice to have source in capital letters
```suggestion
    file.path(outdir, paste(toupper(source), "site", siteid_short, sep = "_"))
```